### PR TITLE
Disposable - Updated `cba_disposable_fnc_replaceMagazineCargo`

### PR DESCRIPTION
**Pull request by johnb432**
**When merged this pull request will:**
- Title.
- Cleaned up `FUNC(replaceMagazineCargo)`.
- Now also checks if launchers can fit in vests and uniform and replaces them if possible.
  I think this case is rare, but it's technically possible so I think CBA should be able to handle it.
- Now handles all containers within containers (previously only backpacks within containers).
- Fixes a bug related to grenades not being throwable (https://feedback.bistudio.com/T74244), as this now only removes the necessary magazines and not every magazine.

### CODE REVIEW

File: addons/disposable/fnc_replaceMagazineCargo.sqf [hunk 1]
```diff
@@ -6,61 +6,41 @@ Description:
...
-};
+if (!local _container) exitWith {};
+if (missionNamespace getVariable [QGVAR(disableMagazineReplacement), false]) exitWith {};
 
-private _backpackContainer = backpackContainer _box;
-if (!isNull _backpackContainer) then {
-    _backpackContainer call FUNC(replaceMagazineCargo);
-};
+private _containers = everyBackpack _container;
+_containers append ([uniformContainer _container, vestContainer _container, backpackContainer _container] select {!isNull _x});
```
Comment (PabstMirror) [inline addons/disposable/fnc_replaceMagazineCargo.sqf:31]:
```suggestion
if (_container isKindOf "CaManBase") then {
    _containers append ([uniformContainer _container, vestContainer _container, backpackContainer _container] select {!isNull _x});
};
```
possibly, I'll test this later

**Comment by PabstMirror** [comment]
looks good overall, I don't understand the `"isBackpack") != -1;` either
also wondering if `addWeaponCargoGlobal` will ignore slot restrictions, will test in a bit

**Comment by johnb432** [comment]
> also wondering if `addWeaponCargoGlobal` will ignore slot restrictions, will test in a bit

It does - I'm guessing that code was to stop that then.
Do we want to keep the current behaviour? I don't like that stuff vanishes, just because it isn't allowed to be in a backpack. I'd prefer if it were dropped on the ground, but that become quite messy.

**Comment by PabstMirror** [comment]
loadout for b_lat:
2 nlaws in the backpack (nlaws can't normally fit in backpacks)


But I agree that this is better than having things disappear
i.e. in vanilla you would get 3 shots, now you still have 3 shots

I guess we could always look at a setting if people really have a problem with it? (I'd rather not have to)

**Review by PabstMirror — APPROVED** [review]

**Comment by Drofseh** [comment]
I think you should always get the number of shots the mission maker intended. So fill up the backpack if required.

**Comment by johnb432** [comment]
As currently nobody wants to change it, but *if* in future we want to modify this PR, then I would not revert the code to the old stuff. It didn't take into account if launchers could fit in uniforms or vests, which is probably rare but not impossible.

**Comment by johnb432** [comment]
Discussion in discord (https://discord.com/channels/976165959041679380/976228110078992456/1237139353650597908) concluded that the PR in its original form was not wanted. I made a new PR #1664 where only the fix is present.

This PR now only brings the code up-to-date and also checks if launchers can fit in uniforms and vests and replaces them if possible.

**Comment by PabstMirror** [comment]
pushed a small change allowing us to keep the same args for `cba_disposable_fnc_replaceMagazineCargo`
my worry is someone might be doing something like
- add loadout to crate
- call cba_disposable_fnc_replaceMagazineCargo

and the change in args might cause problems (even though it says `Internal Function`)

**Review by PabstMirror — APPROVED** [review]

**Comment by johnb432** [comment]
> and the change in args might cause problems (even though it says `Internal Function`)

I really feel that's a "them" problem.
If you want the changes to remain, I'll need to cleanup some leftover stuff.

**Comment by johnb432** [comment]
> I really feel that's a "them" problem. If you want the changes to remain, I'll need to cleanup some leftover stuff.

The reason I added both the container and its type is because it doesn't work for uniforms and vests. If you do `typeOf uniformContainer player`, it returns something like `"Supply40"`, which can't be used to determine if a launcher is supposed to fit in a uniform/vest or not.

Usually launchers don't fit in uniforms and vests, but I'd prefer to be thorough and take every possibility into account.

**Pull request merged**